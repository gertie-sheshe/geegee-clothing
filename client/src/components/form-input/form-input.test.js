import React from 'react';
import { shallow } from 'enzyme';

import FormInput from './form-input.component';

describe('FormInput Component Tests', () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: 'email',
      value: 'test@gmail.com',
      handleChange: mockHandleChange,
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  it('Should render FormInput component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call handleChange method when input changes', () => {
    wrapper.find('FormInputContainer').simulate('change');

    expect(mockHandleChange).toHaveBeenCalled();
  });

  it('Should render FormInputLabel if there is a label', () => {
    expect(wrapper.exists('FormInputLabel')).toBe(true);
  });

  it('Should not render FormInputLabel if there is no label', () => {
    const newMockProps = {
      label: '',
      value: 'test@gmail.com',
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormInput {...newMockProps} />);
    expect(newWrapper.exists('FormInputLabel')).toBe(false);
  });
});
