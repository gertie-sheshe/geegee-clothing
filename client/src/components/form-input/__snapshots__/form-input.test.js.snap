// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component Tests Should render FormInput component 1`] = `ShallowWrapper {}`;
